### PR TITLE
Add variant for OpenMP runtime

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xhochy** [comment]
@igorsafo Any idea how I could check that the correct variant is used? In both cases it seems TBB is linked but I would have only expected that it would be linked in one case.

Also, should we also set `DNNL_CPU_THREADING_RUNTIME`?

**Comment by igorsafo** [comment]
Hi @xhochy , thank you for your contribution!

I think the reason tbb is always linked is the following part of build_onednn.sh:
```sh
export TBBROOT=${PREFIX}
export LDFLAGS="${LDFLAGS} -ltbb"
if [[ "${dnnl_cpu_runtime}" == "tbb" ]]; then
  DNNL_CPU_RUNTIME="TBB"
elif [[ "${dnnl_cpu_runtime}" == "omp" ]]; then
  DNNL_CPU_RUNTIME="OMP"
fi
```
I don't think we need to add '-ltbb' to LDFLAGS, because  it will be done by oneDNN [build system](https://github.com/oneapi-src/oneDNN/blob/1a669997abc3ec8b2ed0d842923fe43a61e05adb/CMakeLists.txt#L114).

`DNNL_CPU_THREADING_RUNTIME` is an internal variable, which will be set automatically by oneDNN build system based on `DNNL_CPU_RUNTIME`:
```
# file: cmake/Threading.cmake
# CPU threading runtime specifies the threading used by the library:
# sequential, OpenMP or TBB. In future it may be different from CPU runtime.
if(DNNL_CPU_SYCL)
    set(DNNL_CPU_THREADING_RUNTIME "TBB")
else()
    set(DNNL_CPU_THREADING_RUNTIME "${DNNL_CPU_RUNTIME}")
endif()
```

Regards,
Igor Safonov

**Comment by xhochy** [comment]
@conda-forge-admin please rerender

**Comment by xhochy** [comment]
@igorsafo Can you have another look at this?

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+dnnl_cpu_runtime:
+  - tbb
```
Comment (igorsafo) [inline recipe/conda_build_config.yaml:2]: (random spot) Do I understand correctly that you changed only `recipe/*` files while other we re-generated automatically?

Comment (xhochy) [reply recipe/conda_build_config.yaml:2]: Yes!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -14,7 +12,7 @@ source:
     - 1009.patch
 
 build:
-  number: 0
+  number: 1
```
[the remark this answers is not in the document]
Comment (xhochy) [reply recipe/meta.yaml:15]: I increased it here as I added the `track_features` below.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -25,7 +23,10 @@ outputs:
       string: {{ dnnl_cpu_runtime }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
       run_exports:
         - {{ pin_subpackage("onednn", max_pin="x") }}
-      # TODO: Add track_features once we support more than a single variant
+      {% if dnnl_cpu_runtime != "omp" %}
+      track_features:
```
[the remark this answers is not in the document]
Comment (xhochy) [reply recipe/meta.yaml:27]: We use it here to add a feature to all-but-OpenMP. This is simply a trick to set a default variant that is installed if no explicit was selected. `conda` minimizes the tracked features across all packages when computing a solution, thus OpenMP is selected by default for oneDNN.

File: recipe/build-onednn.sh [hunk 4]
```diff
@@ -4,10 +4,15 @@ set -exuo pipefail
 
 mkdir build
 pushd build
-export TBBROOT=${PREFIX}
-export LDFLAGS="${LDFLAGS} -ltbb"
+if [[ "${dnnl_cpu_runtime}" == "tbb" ]]; then
+  export TBBROOT=${PREFIX}
+  export LDFLAGS="${LDFLAGS} -ltbb"
```
[the remark this answers is not in the document]
Comment (xhochy) [reply recipe/build-onednn.sh:9]: I'll push a revision without this line and then let's see whether it works.

**Review by igorsafo — APPROVED** [review]
The changes LGTM, thank you!

File: recipe/build-onednn.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (igorsafo) [reply recipe/build-onednn.sh]: Sounds good to me, thanks

**Pull request merged**